File: src/2025-03-04-03-28-40--deepseek-chat-0304/javascript/exercises/practice/transpose/transpose.js
export const transpose = (input) => {
  if (input.length === 0) return [];
  
  // Find the maximum line length
  const maxLength = Math.max(...input.map(line => line.length));
  
  // Pad lines with spaces on the right to make them equal length
  const padded = input.map(line => line.padEnd(maxLength, ' '));
  
  // Transpose the matrix
  const transposed = [];
  for (let i = 0; i < maxLength; i++) {
    let column = '';
    for (let j = 0; j < padded.length; j++) {
      column += padded[j][i];
    }
    transposed.push(column);
  }
  
  // Remove only trailing spaces that were added by padding
  return transposed.map(line => {
    // Find the last non-space character
    const lastCharIndex = line.split('').findLastIndex(c => c !== ' ');
    // If the line is all spaces, return empty string
    if (lastCharIndex === -1) return '';
    // Return the line up to and including the last non-space character
    return line.slice(0, lastCharIndex + 1);
  });
};

File: src/2025-03-04-03-28-40--deepseek-chat-0304/java/exercises/practice/transpose/src/main/java/Transpose.java
public class Transpose {
    public String transpose(String toTranspose) {
        if (toTranspose.isEmpty()) {
            return "";
        }

        String[] lines = toTranspose.split("\n");
        int maxLength = 0;
        
        // Find the maximum line length
        for (String line : lines) {
            if (line.length() > maxLength) {
                maxLength = line.length();
            }
        }

        // Build the transposed result
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < maxLength; i++) {
            StringBuilder row = new StringBuilder();
            for (String line : lines) {
                if (i < line.length()) {
                    row.append(line.charAt(i));
                } else if (i < maxLength) {
                    row.append(' ');
                }
            }
            // Remove trailing spaces from the row
            int lastChar = row.length() - 1;
            while (lastChar >= 0 && row.charAt(lastChar) == ' ') {
                lastChar--;
            }
            if (lastChar >= 0) {
                row.setLength(lastChar + 1);
            }
            if (i < maxLength - 1) {
                result.append(row).append("\n");
            } else {
                result.append(row);
            }
        }

        return result.toString();
    }
}
